### PR TITLE
Counting steps

### LINKED ISSUE #23: steps_to_floor

### PR DESCRIPTION
**Pull request by Elshikh-Amro**
---
name: solution review 
about: A template PR for code review with a checklist
---

### Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
  - [x] There is no logic in the unit test
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [x] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [x] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [x] Each assumption about an argument is checked with an assertion
  - [x] Each assertion checks for _only one_ assumption about the argument
- [x] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [x] The function has correct type annotations

### Do's

- [x] Variable names help to understand the strategy
- [x] Any comments are clear and describe the strategy
- [x] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [x] The function's strategy _is not_ described in the documentation
- [x] Comments explain the _strategy_, **not** the _implementation_
- [x] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [x] The code passes the formatting checks
- [x] The code passes all Ruff linting checks
- [x] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [x] Variables are named with snake_case
- [x] Variable names are clear and helpful
- [x] The code follows the strategy as simply as possible
- [x] The implementation is as simple as possible given the strategy
- [x] The code includes defensive assertions
- [x] Defensive assertions include as little logic as possible

### CODE REVIEW

File: solutions/steps_to_floor.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    """
+    Calculates the number of steps required to reach a given floor in a building,
+    assuming it takes 10 steps to move between consecutive floors.
+
+    If no argument is provided, the function will prompt the user for input.
+
+    Parameters:
+      target_floor: int (optional), the floor number to reach (positive, negative, or zero).
+
+    Returns -> int, the total number of steps required.
```
Comment (CaesarGhazi) [inline solutions/steps_to_floor.py]: i believe here after the `Returns` there should be the defensive assertions documented using `Raises: `

Comment (CaesarGhazi) [reply solutions/steps_to_floor.py]: and right after that it should contain the doctests. 
such as: 
```
>>>steps_to_floor(0)
0
```

File: solutions/tests/test_steps_to_floor.py
Comment (CaesarGhazi) [inline solutions/tests/test_steps_to_floor.py:1]: i think here there should be docstrings for the function and every test case.
and every test case should include one assertion.

File: solutions/tests/test_steps_to_floor.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    def test_positive_floor(self):
+        self.assertEqual(steps_to_floor(1), 10)
+        self.assertEqual(steps_to_floor(5), 50)
+
+    def test_negative_floor(self):
+        self.assertEqual(steps_to_floor(-1), 10)
+        self.assertEqual(steps_to_floor(-3), 30)
+
+    def test_invalid_input(self):
```
Comment (CaesarGhazi) [inline solutions/tests/test_steps_to_floor.py:28]: these are defensive test cases i'd recommend adding a comment to show thatm it'll help alot.

File: solutions/tests/test_steps_to_floor.py
Comment (CaesarGhazi) [reply solutions/tests/test_steps_to_floor.py:1]: i'd also suggest adding Edge cases tests

File: solutions/steps_to_floor.py (hunk 1 above)
Comment (Elshikh-Amro) [reply solutions/steps_to_floor.py]: Raises:
      AssertionError: If the target_floor is not an integer.
      ValueError: If the user input is not a valid integer.

    Examples:
    >>> steps_to_floor(0)
    0
    >>> steps_to_floor(1)
    10
    >>> steps_to_floor(5)
    50
    >>> steps_to_floor(-3)
    30
    """
like this you mean?

**Review by CaesarGhazi — CHANGES_REQUESTED** [review]
the code looks good, it's well formated. 
GREATJOB!!!
just one thing, on the test file, i think here there should be docstrings for the function and every test case.

**Comment by Elshikh-Amro** [comment]
ok @CaesarGhazi can you check again

**Review by CaesarGhazi — APPROVED** [review]
this looks amazing, it's well formated and adheres to the code checklist.
GREATJOOBBB!!!
